src/overlays/effects/ovl_Effect_Ss_Dust/z_eff_ss_dust.c

```diff
@@ -1,10 +1,196 @@
-#include <ultra64.h>
-#include <global.h>
+/*
+ * File: z_eff_ss_dust.c
+ * Overlay: ovl_Effect_Ss_Dust
+ * Description: Dust Particle Effect
+ */
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/effects/ovl_Effect_Ss_Dust/func_809A22D0.s")
+#include "z_eff_ss_dust.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/effects/ovl_Effect_Ss_Dust/func_809A2480.s")
+u32 EffectSsDust_Init(GlobalContext* globalCtx, u32 index, EffectSs* this, void* initParamsx);
+void EffectSsDust_Update(GlobalContext* globalCtx, u32 index, EffectSs* this);
+void EffectSsDust_Unused(GlobalContext* globalCtx, u32 index, EffectSs* this);
+void EffectSsDust_Draw(GlobalContext* globalCtx, u32 index, EffectSs* this);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/effects/ovl_Effect_Ss_Dust/func_809A27F0.s")
+EffectSsInit Effect_Ss_Dust_InitVars = {
+    EFFECT_SS_DUST,
+    EffectSsDust_Init,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/effects/ovl_Effect_Ss_Dust/func_809A28EC.s")
+static void* sUpdateFuncs[] = { EffectSsDust_Update, EffectSsDust_Unused };
+
+UNK_PTR D_809A2A50[] = {
+    0x04051DB0, 0x040521B0, 0x040525B0, 0x040529B0, 0x04052DB0, 0x040531B0, 0x040535B0, 0x040539B0
+};
+
+typedef enum {
+    /* 0x00 */ SS_DUST_PRIM_R,
+    /* 0x01 */ SS_DUST_PRIM_G,
+    /* 0x02 */ SS_DUST_PRIM_B,
+    /* 0x03 */ SS_DUST_PRIM_A,
+    /* 0x04 */ SS_DUST_ENV_R,
+    /* 0x05 */ SS_DUST_ENV_G,
+    /* 0x06 */ SS_DUST_ENV_B,
+    /* 0x07 */ SS_DUST_ENV_A,
+    /* 0x08 */ SS_DUST_TEX_IDX, // this reg is also used to set specific colors in the unused update function
```

Comment:
Like I said in the previous comment, it is actually unused in the final game, and its important to clarify because its used for a completely different purpose in the main update function than it is in the unused one. But if theres a better way to word that let me know

Reply:
Yeah I'm not saying the comment isn't relevant, I just think the update mode should be referred by something other than "unused".

Reply:
right. I tend to look at naming things through the perspective of preserving and documenting, and often forget that someone may decide to use this in the future for modding purposes. Ill try my best to remember that when naming things but let me know when it happens FeelsOkayMan